cloudinit/distros/ug_util.py

```diff
@@ -16,86 +16,69 @@
 LOG = logging.getLogger(__name__)
 
 
-# Normalizes a input group configuration
-# which can be a comma seperated list of
-# group names, or a list of group names
-# or a python dictionary of group names
-# to a list of members of that group.
+# Normalizes an input group configuration which can be:
+# Comma seperated string or a list or a dictionary
 #
-# The output is a dictionary of group
-# names => members of that group which
-# is the standard form used in the rest
-# of cloud-init
+# Returns dictionary of group names => members of that group which is the
+# standard form used in the rest of cloud-init
 def _normalize_groups(grp_cfg):
+    groups = {}
```

Comment:
Nit: I think it makes sense to keep groups where it was. It doesn't get used until right after where it was defined, so it makes sense to keep the declaration close.